swan/python/jupyter_server_config.py

```diff
@@ -3,7 +3,19 @@
 
 home = os.environ.get("HOME")
 jupyter_path = os.environ.get("JUPYTER_PATH", f"{home}/.local/share/jupyter")
-site_packages_path = getsitepackages()[0]
+site_packages_paths = getsitepackages()
+
+# Find the first site-packages path containing the desired template folder
+template_path = None
+for path in site_packages_paths:
+    candidate = os.path.join(path, "swancustomenvironments", "templates")
+    if os.path.isdir(candidate):
+        template_path = candidate
+        break
+
+# Fallback or error if not found
```

Comment:
I would place all this code down together with the definition of `extra_template_paths`.